File: src/main/java/moe/zhi/controller/GreetingResource.java
package moe.zhi.controller;

import io.quarkus.cache.CacheResult;

import javax.ws.rs.GET;
import javax.ws.rs.Path;
import javax.ws.rs.Produces;
import javax.ws.rs.core.MediaType;

@Path("/hello")
public class GreetingResource {

    @CacheResult(cacheName = "hello-cache")
    @GET
    @Produces(MediaType.TEXT_PLAIN)
    public String hello() {
        return "Hello from RESTEasy";
    }
}

File: src/main/java/moe/zhi/controller/GreetingResource.kt
package moe.zhi.controller

import io.quarkus.cache.CacheResult
import jakarta.ws.rs.GET
import jakarta.ws.rs.Path
import jakarta.ws.rs.Produces
import jakarta.ws.rs.core.MediaType

@Path("/hello")
class GreetingResource {
    @CacheResult(cacheName = "hello-cache")
    @GET
    @Produces(MediaType.TEXT_PLAIN)
    fun hello(): String {
        return "Hello from RESTEasy"
    }
}